providers/src/airflow/providers/amazon/aws/operators/redshift_data.py

```diff
@@ -124,12 +126,11 @@ def __init__(
                 poll_interval,
             )
         self.return_sql_result = return_sql_result
-        self.statement_id: str | None = None
         self.deferrable = deferrable
         self.session_id = session_id
         self.session_keep_alive_seconds = session_keep_alive_seconds
 
-    def execute(self, context: Context) -> GetStatementResultResponseTypeDef | str:
+    def execute(self, context: Context) -> list[GetStatementResultResponseTypeDef] | list[str]:
```

Comment:
This is a breaking change I guess we can consider it as a bug fix

Reply:
Agreed, part of the change that needed to be made.